app/javascript/bundles/RestaurantsApp/components/RestaurantsApp.jsx

```diff
@@ -60,6 +67,11 @@ export default class RestaurantsApp extends React.Component {
         this.afterOpenModal = this.afterOpenModal.bind(this);
         this.closeModal = this.closeModal.bind(this);
         this.onRestaurantsChange = this.onRestaurantsChange.bind(this);
+        this.clearFilters = this.clearFilters.bind(this);
+        this.loadGoogleMap = this.loadGoogleMap.bind(this);
+        this.onRestaurantClick = this.onRestaurantClick.bind(this);
+        this.onRestaurantEnter = this.onRestaurantEnter.bind(this);
+        this.onRestaurantLeave = this.onRestaurantLeave.bind(this);
```

Comment:
Seems to me like a good indicator there's some refactoring in order. :)

Reply:
For example, there's no apparent (that I can see) need for `loadGoogleMap` and `loadJS` to be in the component.
You can simply extract them out of the component into their own functions, or even their own file.

Reply:
I refactored according to this article: https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56
Using the 5th method, defining the function as an arrow function